src/main/scala/io/phdata/retirementage/loadgen/LoadGenerator.scala

```diff
@@ -1,39 +1,194 @@
+      */
+    val conf = new LoadGeneratorConfig(args)
+
+    val spark = SparkSession
+      .builder()
+      .enableHiveSupport()
+      .appName("load-generator")
+      .config("spark.sql.parquet.compression.codec", "snappy")
+      .config("spark.sql.parquet.binaryAsString", "true")
+      .config("spark.serializer", "org.apache.spark.serializer.KryoSerializer")
+      .getOrCreate()
+
     generateTables(spark, conf)
   }
 
   def generateTables(spark: SparkSession, conf: LoadGeneratorConfig): Unit = {
+    // test stuff
+    val coalNum = conf.factCount() / 250000
+
     // generate fact table
+    val factdf = generateTable(spark, conf.factCount(), 1)
     // write fact table to disk
+    factdf.write
+      .mode(SaveMode.Overwrite)
+      .saveAsTable(s"${conf.databaseName()}.${conf.factName()}")
+
     // generate dimension table
-    // write dimension table to disk
-    // generate subdimension table
-    // write subdimension table to disk
+//    val dimensionDf =
+//      generateTableFromParent(spark, conf.dimensionCount(), 3, conf.factCount(), factdf)
+//    // write dimension table to disk
+//    dimensionDf.write
+//      .mode(SaveMode.Overwrite)
+//      .saveAsTable(s"${conf.databaseName()}.${conf.dimName()}")
+//    // generate subdimension table
+//    val subDimensionDf = generateTableFromParent(spark,
+//                                                 conf.subDimensionCount(),
+//                                                 1,
+//                                                 conf.dimensionCount(),
+//                                                 dimensionDf)
+//    // write subdimension table to disk
+//    subDimensionDf.write
+//      .mode(SaveMode.Overwrite)
+//      .saveAsTable(s"${conf.databaseName()}.${conf.subName()}")
   }
 
+  // Creating a case class to use for generateTable
+  case class LoadGenTemp(id: String, payload: String, dimensionid: String, expirationdate: String)
+
   /**
     * Generates a test dataframe
     * @param numRecords Number of records in the dataframe
     * @param payloadBytes Payload size
     * @return The test dataframe
     */
-  def generateTable(numRecords: Int, payloadBytes: Int): DataFrame = ???
+  def generateTable(spark: SparkSession, numRecords: Int, payloadBytes: Int) = {
+
+    /**
+      * Create the following schema:
+      *   -- id: String (nullable = false)
+      *   -- payload: String (nullable = false)
+      *   -- dimensionId: String (nullable = false)
+      *   -- expirationDate: String (nullabe = false)
+      */
+    val schema = StructType(StructField("id", StringType, false) :: Nil)
+      .add(StructField("payload", StringType, false))
+      .add(StructField("dimensionid", StringType, false))
+      .add(StructField("expirationDate", StringType, false))
+    // Create a string with specified bytes
+    val byteString = "a" * payloadBytes
+
+    // Maximum number of records per partition
```

Comment:
Since the word 'partition' is overloaded here (this does not refer to spark partition), you might either define what you mean by partition in a comment or find another word